go/mysql/datetime/parse.go

```diff
@@ -166,24 +166,35 @@ func clampTimeparts(tp *timeparts) bool {
 	if tp.day > 34 || tp.day == 34 && tp.hour > 22 {
 		tp.day = 34
 		tp.hour = 22
-		tp.min = 59
-		tp.sec = 59
+		tp.min = max(tp.min, 59)
+		tp.sec = max(tp.sec, 59)
 		return false
 	}
 	return true
 }
 
-func ParseTime(in string, prec int) (t Time, l int, ok bool) {
+func ParseTime(in string, prec int) (t Time, l int, ok bool, isZero bool) {
 	in = strings.Trim(in, " \t\r\n")
 	if len(in) == 0 {
-		return Time{}, 0, false
+		return Time{}, 0, false, false
 	}
 	var neg bool
 	if in[0] == '-' {
 		neg = true
 		in = in[1:]
 	}
 
+	for i, d := range in {
```

Comment:
Earlier we didn't check if the `TIME` which needs to be returned is zero, so we returned `NULL` before. Returning a bool which states that if the returned value is zero, solves the issue.